### ISSUE TITLE
FR: landscape mode

### ISSUE THREAD

**Issue by irregularisnotrandom** [seizonsenryaku/HayaiLauncher#67]
This app is portrait mode. It should respect the rotation?

**Comment by seizonsenryaku** [comment]
The current portrait-only layout is intentional (the keyboard takes too much space in landscape orientation). Are you using a tablet?

**Comment by irregularisnotrandom** [comment]
Yes i am on a tablet. The porteait only mode is quite awkward if you will

**Comment by irregularisnotrandom** [comment]
any word on this? I love this app but it is abit pain to use because of portrait only mode.

**Comment by seizonsenryaku** [comment]
I'm too busy to work on this right now, sorry.

**Comment by Nowaker** [comment]
I have a qwerty phone (Motorola Photon Q), whenever I open the keyboard it's expected that the screen rotates.

**Comment by staviss** [comment]
Submitting.
I don't search apps always, sometimes only, the keyboard is hidden by default. And rotating the phone to start an app is confusing.

**Comment by avuton** [comment]
I have [a branch](https://github.com/avuton/HayaiLauncher/tree/landscape) which adds support for landscape mode. It is default disabled (due to legacy reasons), enable it in the settings menu. It is _not_ complete yet. With any layout stuff, it can break in all sorts of crappy ways, depending on device. I need feedback. If I don't get it, I still intend to PR, but it's likely to take quite a bit more time. It is broken in the emulator, since I have tested it on two devices, I'm assuming right now that it's buggy emulation.

It may change often and rapidly. I often delete history, so, commits disappear at a whim. With that said, please give it a shot, give some feedback and I'll try to get a PR in sooner than later. 👍

**Comment by koiuo** [comment]
@avuton I was going to try implementing that too. I'm glad, I've found your message, and it's only 6 days old :) Thanks for your work!

As for the stability, there's a regression caused by your re-factoring.

You call `mSearchEditText.clearFocus()` if _Open keyboard automatically_ is set to `false`. Documentation says, that element will _try_ to give up focus giving to the first available element, however, if it is the first element in container, it may receive it again. And that's what is happening, I suspect: search field always has bilking cursor in it even if keyboard is not opened.

Really looking forward for your changes to get merged.

**Comment by koiuo** [comment]
@avuton 
As a quick fix I added
`mAppsContainer.requestFocus()` instead of `mSearchEditText.clearFocus()`.

This fixes the regression, but overall bug remains (present in upstream version too): after keyboard is hidden, search field remains focused. 
I reported it as [issues/103](https://github.com/seizonsenryaku/HayaiLauncher/issues/103). Perhaps you could address that as well in your PR.

**Comment by avuton** [comment]
Wow, thanks for the testing. I've added your suggestion in `landscape`, my fix might be slightly more comprehensive and it ~~could fix #103 as well~~. I would have never noticed this regression, didn't pop up in my head as important, so I couldn't thank you more. **It did not fix 103**. Not sure I'll get around to it, that's not (yet) my motivation.

**Comment by eevleevs** [comment]
@avuton: landscape works perfectly on Samsung A320F, thanks!
